src/uwtools/config/formats/ini.py

```diff
@@ -0,0 +1,94 @@
+
+    def __init__(
+        self,
+        config_file: str,
+        space_around_delimiters: bool = True,
+    ):
+        """
+        Construct an INIConfig object.
+
+        Spaces may be included for INI format, but should be excluded for bash.
+
+        :param config_file: Path to the config file to load.
+        :param space_around_delimiters: Include spaces around delimiters?
+        """
+        super().__init__(config_file)
+        self.space_around_delimiters = space_around_delimiters
+        self.parse_include()
+
+    # Private methods
+
+    def _load(self, config_file: OptionalPath) -> dict:
+        """
+        Reads and parses an INI file.
+
+        See docs for Config._load().
+
+        :param config_file: Path to config file to load.
+        """
+        # The protected _sections method is the most straightforward way to get at the dict
+        # representation of the parse config.
+
+        cfg = configparser.ConfigParser()
+        cfg.optionxform = str  # type: ignore
+        sections = cfg._sections  # type: ignore # pylint: disable=protected-access
+        with readable(config_file) as f:
+            raw = f.read()
+        try:
+            cfg.read_string(raw)
+            return dict(sections)
+        except configparser.MissingSectionHeaderError:
+            cfg.read_string("[top]\n" + raw)
+            return dict(sections.get("top"))
+
+    # Public methods
+
+    def dump(self, path: OptionalPath) -> None:
+        """
+        Dumps the config in INI format.
+
+        :param path: Path to dump config to.
+        """
+        INIConfig.dump_dict(path, self.data, ns(space=self.space_around_delimiters))
+
+    @staticmethod
+    def dump_dict(path: OptionalPath, cfg: dict, opts: Optional[ns] = None) -> None:
+        """
+        Dumps a provided config dictionary in INI or sh format.
+
+        :param path: Path to dump config to.
+        :param cfg: The in-memory config object to dump.
+        :param space_around_delimiters: Place spaces around delimiters?
+        """
+        # Configparser adds a newline after each section, presumably to create nice-looking output
+        # when an INI contains multiple sections. Unfortunately, it also adds a newline after the
+        # _final_ section, resulting in an anomalous trailing newline. To avoid this, write first to
+        # memory, then strip the trailing newline.
+        parser = configparser.ConfigParser()
+        s = StringIO()
+        cfgdepth = depth(cfg)
+        assert cfgdepth in (1, 2)  # 1 => .sh, 2 => .ini
+        if cfgdepth == 2:
+            parser.read_dict(cfg)
+            parser.write(s, space_around_delimiters=opts.space if opts else True)
+        else:
+            space = " " if not opts or opts.space else ""
+            for key, value in cfg.items():
+                print(f"{key}{space}={space}{value}", file=s)
+        with writable(path) as f:
+            print(s.getvalue().strip(), file=f)
+        s.close()
```

Comment:
Also, I think much of this would be moot and could be simplified if/when we create a `ShConfig` class to draw a distinction between ini and sh. I'll be creating a ticket for that.